Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -1682,13 +1679,17 @@ PhysicalParticleContainer::AddPlasmaFlux (amrex::Real dt)
                     continue;
                 }
 #endif
-
-                // Save the x and y values to use in the insideBounds checks.
-                // This is needed with WARPX_DIM_RZ since x and y are modified.
-                Real xb = ppos.x;
-                Real yb = ppos.y;
+                // Lab-frame simulation
+                // If the particle is not within the species's
+                // xmin, xmax, ymin, ymax, zmin, zmax, go to
+                // the next generated particle.
+                if (!inj_pos->insideBounds(ppos.x, ppos.y, ppos.z)) {
```

Comment:
This check is now done **before** calling `theta = 2._prt*MathConst::pi*amrex::Random(engine);` a few lines below. This will therefore change the random number generation, and probably explains why I had to reset the benchmarks.